src/core/Core.hpp

```diff
@@ -31,13 +31,14 @@ class Core{
     std::uint32_t minQuantity;
     std::uint32_t maxQuantity;
 
-    std::uint32_t getProductCurrentQuantity() const;
-    std::uint32_t getProductMinQuantity() const;
-    std::uint32_t getProductMaxQuantity() const;
-    MethodResult changeProductQuantity(const std::string &option, const std::uint32_t quantity);
+    std::uint32_t getProductCurrentQuantity(const std::uint32_t &productID);

```

Comment:
https://stackoverflow.com/questions/14013139/is-it-counter-productive-to-pass-primitive-types-by-reference

Przekazywanie prostych typów przez referencje nie jest czyms optymalnym. Referencje służa nam szczególnie to przekazywanie większych obiektow, np jak klasy, stringi czy vectory.

Tutaj przekazuje const uint32_t&. Z racji tego, że to const, to już wiem, że nie będzie ta wartość modyfikowana. Z tego powodu, i z racji tego że to typ prostu, & jest tutaj zbędna